ui/widgets/nuxeo-dialog.js

```diff
@@ -89,6 +89,7 @@ IronOverlayManager._overlayWithBackdrop = function() {
     ready() {
       super.ready();
       this.addEventListener('iron-overlay-opened', this._opened);
+      this.addEventListener('iron-overlay-closed', this._teardownInstanceChildren);
```

Comment:
Any particular reason why we need to run thins when the overlay is closed? It seems to me that the issue is caused by `disconnectedCallback` being called when the dropdown menu is closed, which makes sense. However, on the `disconnectedCallback`, we are just doing `this._instance = null;`, which is not enough, looking at other elements that also follow the same pattern (see [nuxeo-filter](https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-filter.js#L292-L310)). I think we could do the same here: borrow the `_clear` method (which is basically your `_teardownInstanceChildren` method) and call it on the `disconnectedCallback`. Because we are setting `_instance ` to null there, we already know that those contents will be replaced anyway when the dialog is open, and this way we won't need to have another variable,

Reply:
You are totally right. In fact I tried to follow the same pattern we have in `nuxeo-filter`, but for some reason when I tried to access the `this._instance.children` in the `disconnectedCallback` the collection was already empty, and there were no children nodes to remove. 
This is the reason why I saved the `this._instance.children` in a local variable, right after appending the instance root.

Reply:
So far it works, not sure why it wasn't working before :/ Thanks Gabe.